pygeos/io.py

```diff
@@ -239,6 +239,13 @@ def from_shapely(geometry, **kwargs):
         arr = np.empty(1, dtype=object)
         arr[0] = geometry
         arr.shape = ()
+    elif geometry is None:
+        arr = geometry
+    elif not isinstance(geometry, np.ndarray):
+        # geometry is a list/array-like
+        arr = np.empty(len(geometry), dtype=object)
+        arr[:] = geometry
```

Comment:
@caspervdw this fails for the `ShapelyGeometryMock` that we use in our tests (because this is not a subclass of shapely's `BaseGeometry`) .. Would you be fine to have those tests actually use Shapely instead? 
I am not fully sure how to make it work otherwise (unless I would make the above `isinstance(geometry, ShapelyGeometry)` or duck-type check like `hasattr(geometry, "__geom__")`, but then that change would *only* be to please the mock in the tests ..

Reply:
Actually I needed a fix for this for the shapely dev branch as well, since I am using a DummyGeometry class there to pass pointers to GEOS, which is also not a BaseGeometry subclass. So added an additional check for the object having a length (being a Collection abc)

Reply:
I think that it is sufficient to skip the unittests if shapely is not present.

Reply:
I think it was actually fixed now by adding the additional check for being a collection here (so the ShapelyGeometryMock then passes through to the next `else:` block)